pkg/prometheus/promcfg.go

```diff
@@ -2737,6 +2737,10 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 		cfg = cg.WithMinimumVersion("2.49.0").AppendMapItem(cfg, "enable_compression", *sc.Spec.EnableCompression)
 	}
 
+	if sc.Spec.EnableHTTP2 != nil {
+		cfg = cg.AppendMapItem(cfg, "enable_http2", *sc.Spec.EnableHTTP2)
```

Comment:
sorry I missed this in my previous review but we need to check the Prometheus version.
```suggestion
		cfg = cg.WithMinimumVersion("2.35.0").AppendMapItem(cfg, "enable_http2", *sc.Spec.EnableHTTP2)
```